cwl/src/main/scala/cwl/CwlExpressionCommandPart.scala

```diff
@@ -43,3 +44,29 @@ case class CwlArgumentCommandPart(argument: CommandLineBinding) extends CommandP
   }
 }
 
+case class InputParameterCommandPart(argument: CommandInputParameter) extends CommandPart {
+
+  override def instantiate(inputsMap: Map[LocalName, WomValue],
+                           functions: IoFunctionSet,
+                           valueMapper: (WomValue) => WomValue) = {
+
+    val localInputsMap = inputsMap.map{
+      case (LocalName(localName), WomSingleFile(path)) => localName -> WomString(path)
+      case (LocalName(localName), value) => localName -> value
+    }
+
+
+    val womValue: WomValue = argument match {
```

Comment:
At first I thought this was a straight copy/paste from above, but it does seem to be fundamentally different. 

Since it's so hard to parse the chains of `Some`s and `None`s and `_`s, could you add comments on what situation each case is trying to hit?